### PR TITLE
Colpkg

### LINKED ISSUE #5107: Switch from 'collection.apkg' to new '*.colpkg'
###### Reproduction Steps

1. Export a collection  --> default name is "collection.apkg"
2. Choose save to file option
3. Choose a non-default name for the file
4. Import it

###### Expected Result
AnkiDroid recognizes it as a collection rather than a deck

###### Actual Result
AnkiDroid follows anki desktop behavior of expecting it to be called exactly "collection.apkg" in order to be recognized as a collection.

###### Possible solutions
We could change the UI to use `ACTION_OPEN_DOCUMENT_TREE` for collections to get a directory to save the file instead of allowing the user to specify the directory and filename, but then we have to show an overwrite confirmation dialog as well, and not let them store multiple collection backups in a single directory.

Rather than perpetuating this unintuitive design, I'd prefer for us to implement an idea I previously suggested to @dae quite sometime ago where we introduce a new file extension / mime type to the anki ecosystem for collections (e.g. call it `*.apkgc` or `*.acol`) instead of requiring those files to be called "collection.apkg", and we solve a bunch of UX problems.

@dae would you be willing to proceed with adding support for this to the non-AnkiDroid clients within a timeframe of say 1 - 2 months?

### PR DESCRIPTION
**Pull request by timrae**
## Purpose / Description
See #5107 and dae/anki@0c80b545

Note: during testing I also found two issues which are fixed in additional commits on this branch
1) crash opening note editor on API16
2) no way to save exported files to filesystem on API16

## Fixes
Fixes #5107

## Approach
Change the extension for collection export to ".colpkg", and allow these files to be imported

## How Has This Been Tested?
On API 28 and API 16 emulators:

* Export deck --> still goes to .apkg file
* Export collection --> now goes to .colpkg file
* import apkg deck above (via file manager and AnkiDroid menu) --> still imports in "add" mode
* import colpkg collection above (via file manager and AnkiDroid menu) --> imports in "replace" mode as expected
* Rename colpkg file above to "collection.apkg" and import --> imports in "replace" mode as expected

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1129,10 +1129,10 @@ private void populateEditFields(String[][] fields, boolean editModelMode) {
             View editline_view = getLayoutInflater().inflate(R.layout.card_multimedia_editline, null);
             FieldEditText newTextbox = (FieldEditText) editline_view.findViewById(R.id.id_note_editText);
 
-            // Use custom implementation of ActionMode.Callback customize selection and insert menus
-            ActionModeCallback actionModeCallback = new ActionModeCallback(newTextbox);
-            newTextbox.setCustomSelectionActionModeCallback(actionModeCallback);
             if (Build.VERSION.SDK_INT >= 23) {
+                // Use custom implementation of ActionMode.Callback customize selection and insert menus
+                ActionModeCallback actionModeCallback = new ActionModeCallback(newTextbox);
+                newTextbox.setCustomSelectionActionModeCallback(actionModeCallback);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: this line is duplicated now?

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerExportCompleteDialog.java [hunk 2]
```diff
@@ -32,25 +32,17 @@ public MaterialDialog onCreateDialog(Bundle savedInstanceState) {
...
-                    .negativeText(R.string.export_save_button)
-                    .onNegative((dialog, which) -> {
-                        ((DeckPicker) getActivity()).dismissAllDialogFragments();
-                        ((DeckPicker) getActivity()).saveExportFile(exportPath);
-                    });
-        }
-
+                .onNegative((dialog, which) -> {
+                    ((DeckPicker) getActivity()).dismissAllDialogFragments();
+                    ((DeckPicker) getActivity()).saveExportFile(exportPath);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerExportCompleteDialog.java]: I was curious about this - the inability to save at all on API19 prior to this change so I looked at the history - as I'd never remove that intentionally. If my read is correct it was always a share and there was no save option in this file prior !? At any rate, much better with the ability to save...

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java [hunk 3]
```diff
@@ -140,6 +140,18 @@ private static void sendShowImportFileDialogMsg(String path) {
         DialogHandler.storeMessage(handlerMessage);
     }
 
+    public static boolean isCollectionPackage(String filename) {
+        return filename != null && (filename.toLowerCase().endsWith(".colpkg") || filename.equals("collection.apkg"));
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java]: this was going to be my first question "so what about the transition, will it handle the old full collections with .apkg suffixes"? yes it does, sweet

**Comment by mikehardy** [comment]
This is great - I know you mentioned a while back you really wanted this for 2.9 but I wasn't sure if you or anyone would have time. It looks great. The only thing that seemed "real" was the duplicated line in NoteEditor, but that's obviously simple enough.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 1 above)
Comment (timrae) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: No, I thought so too, but if you look carefully one is for insertion, and the other is for selection

**Comment by timrae** [comment]
I haven't merged the colpkg commit into the 2.9 release branch, as I'm a little bit worried it might need more testing, so I'm thinking of maybe a 2.9.1 release. On the other hand, since it's an Anki Desktop compatibility issue, it can almost be considered a bug, and if we put it out now, one week of beta testing might be enough. I'm a bit on the fence.

**Comment by mikehardy** [comment]
I'd ship it with a note on the mailing list to test

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: oh of course! that is close, tricked my eye.

**Pull request merged**